odl/space/ntuples.py

```diff
@@ -1767,55 +1939,103 @@ def inner(self, x1, x2):
         inner : float or complex
             The inner product of the two provided vectors
         """
-        inner = _inner_default(x1 * self.vector, x2)
-        if is_real_dtype(x1.dtype):
-            return float(inner)
+        if self.exponent != 2.0:
+            raise NotImplementedError('No inner product defined for '
+                                      'exponent != 2 (got {}).'
+                                      ''.format(self.exponent))
+        else:
+            inner = _inner_default(x1 * self.vector, x2)
+            if is_real_dtype(x1.dtype):
+                return float(inner)
+            else:
+                return complex(inner)
+
+    def norm(self, x):
+        """Calculate the vector-weighted norm of a vector.
+
+        Parameters
+        ----------
+        x : `Fn.Vector`
+            Vector whose norm is calculated
+
+        Returns
+        -------
+        norm : float
+            The norm of the provided vector
+        """
+        if self.exponent == 2.0:
+            return super().norm(x)
```

Comment:
Yes it does, and ~~yes there is a test for it~~ crap, I forgot the test. `super` magic :-)

What happens is that the [norm implementation of `_FnWeightingBase`](https://github.com/odlgroup/odl/blob/issue-246__l2discr/odl/space/base_ntuples.py#L471-L487) is called, which refers to `inner`, which in turn uses weighting. I'll write a test and see if this is really what happens :-)